packages/sdk/src/Portfolio/Integrations/Alice.ts

```diff
@@ -0,0 +1,202 @@
+    type: "uint256",
+  },
+] as const;
+
+export type PlaceOrderArgs = {
+  instrumentId: number;
+  isBuyOrder: boolean;
+  quantityToSell: bigint;
+  limitAmountToGet: bigint;
+};
+
+export function placeOrderEncode(args: PlaceOrderArgs): Hex {
+  return encodeAbiParameters(placeOrderEncoding, [
+    args.instrumentId,
+    args.isBuyOrder,
+    args.limitAmountToGet,
+    args.quantityToSell,
+  ]);
+}
+
+export function placeOrderDecode(encoded: Hex): PlaceOrderArgs {
+  const [instrumentId, isBuyOrder, limitAmountToGet, quantityToSell] = decodeAbiParameters(placeOrderEncoding, encoded);
+
+  return {
+    instrumentId,
+    isBuyOrder,
+    limitAmountToGet,
+    quantityToSell,
+  };
+}
+
+//--------------------------------------------------------------------------------------------
+// REFUND ORDER
+//--------------------------------------------------------------------------------------------
+
+export const refundOrder = ExternalPositionManager.makeUse(Action.RefundOrder, refundOrderEncode);
+
+const refundOrderEncoding = [
+  {
+    type: "uint256",
+    name: "orderId",
+  },
+  {
+    type: "uint16",
+    name: "instrumentId",
+  },
+  {
+    type: "bool",
+    name: "isBuyOrder",
+  },
+  {
+    type: "uint256",
+    name: "quantityToSell",
+  },
+  {
+    type: "uint256",
+    name: "limitAmountToGet",
+  },
+  {
+    type: "uint256",
+    name: "timestamp",
+  },
+] as const;
+
+export type RefundOrderArgs = {
+  orderId: bigint;
+  instrumentId: number;
+  isBuyOrder: boolean;
+  quantityToSell: bigint;
+  limitAmountToGet: bigint;
+  timestamp: bigint;
+};
+
+export function refundOrderEncode(args: RefundOrderArgs): Hex {
+  return encodeAbiParameters(refundOrderEncoding, [
+    args.orderId,
+    args.instrumentId,
+    args.isBuyOrder,
+    args.quantityToSell,
+    args.limitAmountToGet,
+    args.timestamp,
+  ]);
+}
+
+export function refundOrderDecode(encoded: Hex): RefundOrderArgs {
+  const [orderId, instrumentId, isBuyOrder, quantityToSell, limitAmountToGet, timestamp] = decodeAbiParameters(
+    refundOrderEncoding,
+    encoded,
+  );
+
+  return {
+    orderId,
+    instrumentId,
+    isBuyOrder,
+    quantityToSell,
+    limitAmountToGet,
+    timestamp,
+  };
+}
+
+//--------------------------------------------------------------------------------------------
+// SWEEP
+//--------------------------------------------------------------------------------------------
+
+export const sweep = ExternalPositionManager.makeUse(Action.Sweep, sweepEncode);
+
+const sweepEncoding = [
+  {
+    type: "uint256[]",
+    name: "orderIds",
+  },
+] as const;
+
+export type SweepArgs = {
+  orderIds: Array<bigint>;
+};
+
+export function sweepEncode(args: SweepArgs): Hex {
+  return encodeAbiParameters(sweepEncoding, [args.orderIds]);
+}
+
+export function sweepDecode(encoded: Hex): SweepArgs {
+  const [orderIds] = decodeAbiParameters(sweepEncoding, encoded);
+
+  return {
+    orderIds: orderIds as bigint[],
```

Comment:
use of `as` is discouraged if there is an alternative 
```suggestion
    orderIds: [...orderIds],
```